plugins/modules/perlmod_install_info.py

```diff
@@ -0,0 +1,383 @@
+#!/usr/bin/python
+
+# Copyright (c) 2023, Jonathan Kamens <jik@kamens.us>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+# pylint: disable=missing-module-docstring
+
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+DOCUMENTATION = r'''
+---
+module: perlmod_install_info
+
+short_description: Determine from where to install Perl modules
+
+requirements:
+    - C(apt-file) executable in search path on systems that use the apt package
+      manager
+    - C(dnf) or C(yum) executable in search path on systems that use the
+      dnf/yum package manager
+    - C(cpanm) executable in search path if you want to be able to search for
+      packages using cpanm
+
+version_added: "7.0.0"
+
+description:
+    - Searches dnf, yum, apt, and/or cpanm to determine the best source from
+      which to install a Perl module.
+    - Prefers the OS repositories over cpanm.
+    - Specify module names as you would specify them with the C(use) command in
+      a Perl script.
+    - Does not actually install modules. Instead, returns information about
+      where they can be installed from, which can be supplied to subsequent
+      tasks to do the actual installation.
+    - Note that this module will not fail by default if it cannot locate a
+      requested module. If you want that behavior, include a C(failed_when)
+      which checks for C(missing) being non-empty.
+
+options:
+    name:
+        description: Specify one or more Perl modules to search for.
+        required: true
+        type: list
+        elements: str
+    try_installed:
+        description: Specify whether to check if modules are already installed
+          and not look elsewhere if they are.
+        type: bool
+        default: true
+    try_dnf:
+        description: Specify whether to check for modules using the dnf package
+          manager. Defaults to true if dnf executable is available.
+        type: str
+        choices:
+        - 'auto'
+        - 'true'
+        - 'false'
```

Comment:
I would suggest not using `true` and `false` because those keywords are interpreted by YAML as `bool` values. Maybe `enabled` and `disabled` or something along those lines?

Reply:
I don't agree with this suggestion. The code as written will work just fine whether the value as passed to the module is a boolean or a str (it's specifically written to accommodate either case) and "true" and "false" make more sense as the values for these options than "enabled" and "disabled" do.